homeassistant/components/point/config_flow.py

```diff
@@ -1,190 +1,54 @@
-        """Create point session and entries."""
-
-        flow = self.hass.data[DATA_FLOW_IMPL][DOMAIN]
-        client_id = flow[CONF_CLIENT_ID]
-        client_secret = flow[CONF_CLIENT_SECRET]
-        point_session = PointSession(
-            async_get_clientsession(self.hass),
-            client_id,
-            client_secret,
-        )
-        token = await point_session.get_access_token(code)
-        _LOGGER.debug("Got new token")
-        if not point_session.is_authorized:
-            _LOGGER.error("Authentication Error")
-            return self.async_abort(reason="auth_error")
-
-        _LOGGER.info("Successfully authenticated Point")
-        user_email = (await point_session.user()).get("email") or ""
-
-        return self.async_create_entry(
-            title=user_email,
-            data={
-                "token": token,
-                "refresh_args": {
-                    CONF_CLIENT_ID: client_id,
-                    CONF_CLIENT_SECRET: client_secret,
-                },
-            },
-        )
-
-
-class MinutAuthCallbackView(HomeAssistantView):
-    """Minut Authorization Callback View."""
-
-    requires_auth = False
-    url = AUTH_CALLBACK_PATH
-    name = AUTH_CALLBACK_NAME
 
-    @staticmethod
-    async def get(request):
-        """Receive authorization code."""
-        hass = request.app[KEY_HASS]
-        if "code" in request.query:
-            hass.async_create_task(
-                hass.config_entries.flow.async_init(
-                    DOMAIN, context={"source": "code"}, data=request.query["code"]
-                )
-            )
-        return "OK!"
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle Minut Point OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    async def async_step_import(self, data: dict[str, Any]) -> ConfigFlowResult:
+        """Handle import from YAML."""
+        return await self.async_step_user()
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(step_id="reauth_confirm")
+        return await self.async_step_user()
+
+    async def async_oauth_create_entry(self, data: dict[str, Any]) -> ConfigFlowResult:
+        """Create an oauth config entry or update existing entry for reauth."""
+        existing_entry = await self.async_set_unique_id(DOMAIN)
+        if existing_entry:
```

Comment:
```suggestion
        if existing_entry := await self.async_set_unique_id(DOMAIN):
```

Reply:
Oh a walrus, first time I see this in real code.